Dockerfile

```diff
@@ -38,6 +42,11 @@ RUN if [ -n "$NODE_COMMIT" ]; then \
         git clone -b $CLIENT_COMMIT $CLIENT_REPO; \
     else \
         git clone -b $CLIENT_GITBRANCH $CLIENT_REPO; \
+    fi \
+    && if [ -n "$SIDECAR_COMMIT" ]; then \
+        git clone $SIDECAR_REPO && cd casper-sidecar && git checkout $SIDECAR_COMMIT && cd ..; \
```

Comment:
I was wondering why other components - node, client - do not use `git checkout` - but it seems to be bug that you fixed for sidecar.